src/realm/string_compressor.cpp

```diff
@@ -302,12 +302,12 @@ int StringCompressor::compare(CompressedStringView& A, CompressedStringView& B)
         StringData sd_b(str_B.data(), str_B.size());
         REALM_ASSERT_DEBUG(sd_a != sd_b);
         if (sd_a < sd_b)
-            return 1;
-        else
             return -1;
+        else
+            return 1;
     }
     // The compressed strings are identical or one is the prefix of the other
-    return B.size - A.size;
+    return A.size - B.size;
```

Comment:
Please make sure the comment below to matches this change.
I'm also not a fan of such implicit unsigned to signed casting because the result is actually a large unsigned number when A.size is < B.size and we rely on the type conversion to do the right thing here. I think it does for all platforms we care about, but if for some reason the return type changed to `int64_t` then this code would be incorrect.

Reply:
Agreed, I can change the comment and maybe add an explicit cast. I don't think we can ever have a platform where `int` matches with `int64_t`, but never say never.